File: app/src/main/java/com/android/ashwiask/tvmaze/db/favouriteshow/ShowDao.java
package com.android.ashwiask.tvmaze.db.favouriteshow;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import io.reactivex.Single;

@Dao
public interface ShowDao {
    @Query("SELECT * FROM favourite_shows")
    Single<List<FavoriteShow>> getAllFavouriteShows();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(FavoriteShow favoriteShow);

    @Delete
    void remove(FavoriteShow favoriteShow);

    @Query("SELECT count(*) FROM favourite_shows where id LIKE :id")
    int isFavouriteShow(long id);
}

File: app/src/main/java/com/android/ashwiask/tvmaze/db/favouriteshow/ShowDao.kt
package com.android.ashwiask.tvmaze.db.favouriteshow

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface ShowDao {
    @Query("SELECT * FROM favourite_shows")
    suspend fun allFavouriteShows(): List<FavoriteShow>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(favoriteShow: FavoriteShow)

    @Delete
    suspend fun remove(favoriteShow: FavoriteShow)

    @Query("SELECT id from favourite_shows")
    suspend fun getFavoriteShowIds(): List<Long>

    @Query("DELETE FROM favourite_shows")
    suspend fun clear()
}
